tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -11388,15 +11388,7 @@ void hlsl::DiagnoseTranslationUnit(clang::Sema *self) {
     const auto *shaderModel =
         hlsl::ShaderModel::GetByName(self->getLangOpts().HLSLProfile.c_str());
 
-    if (shaderModel->IsGS()) {
-      // Validate that GS has the maxvertexcount attribute
-      if (!pEntryPointDecl->hasAttr<HLSLMaxVertexCountAttr>()) {
-        self->Diag(pEntryPointDecl->getLocation(), diag::err_hlsl_missing_attr)
-            << "GS"
-            << "maxvertexcount";
-        return;
-      }
-    } else if (shaderModel->IsHS()) {
+    if (shaderModel->IsHS()) {
       if (const HLSLPatchConstantFuncAttr *Attr =
               pEntryPointDecl->getAttr<HLSLPatchConstantFuncAttr>()) {
         NameLookup NL = GetSingleFunctionDeclByName(
```

Comment:
Shouldn't this code be moved into `DiagnoseHullEntry`?

Reply:
The reason I can't move this block is because pPatchFnDecl is set by this block, and is used below when checking for recursion. If I moved it into DiagnoseHullEntry, I would need to repeat this code here anyways in order to set pPatchFnDecl.

Reply:
I realize that this code also captures the patch constant function for recursion check later.  I think all this needs to be moved somewhere else eventually, since we need the recursion check on all functions (but also don't want to have to pay a cost of re-checking the same call graph over and over again).

Or is the reason this can't be moved also related to timing - maybe the patch constant function won't have been created yet when we are in `DiagnoseEntry`, for example?